From 238020ad7ea3f280bd81a811e6a79f2a3431698a Mon Sep 17 00:00:00 2001
From: David Disseldorp <ddiss@suse.de>
Date: Wed, 21 Jun 2017 01:45:05 +0200
Subject: [PATCH 4/4] rbd: fix and simplify rbd_osd_req_format_rw()
Patch-mainline: Not yet, clustered LIO/RBD
References: bsc#1045213

rbd_osd_req_format_rw() should set osd_req->r_snapid and
osd_req->r_data_offset, but currently doesn't.

With 7c84883adf6 and 7c84883adf6 upstream kernel changes now merged,
rbd_osd_req_format_rw() can be simplified into a straight wrapper around
rbd_osd_req_format_read() and rbd_osd_req_format_write().

Signed-off-by: David Disseldorp <ddiss@suse.de>
Acked-by: Lee Duncan <lduncan@suse.com>
---
 drivers/block/rbd.c | 20 ++++++--------------
 1 file changed, 6 insertions(+), 14 deletions(-)

diff --git a/drivers/block/rbd.c b/drivers/block/rbd.c
index ca874c4e0fe0..6af4c234c9a3 100644
--- a/drivers/block/rbd.c
+++ b/drivers/block/rbd.c
@@ -1896,20 +1896,6 @@ static void rbd_osd_req_callback(struct ceph_osd_request *osd_req)
 		rbd_obj_request_complete(obj_request);
 }
 
-static void rbd_osd_req_format_rw(struct rbd_obj_request *obj_request)
-{
-	struct rbd_img_request *img_request = obj_request->img_request;
-	struct ceph_osd_request *osd_req = obj_request->osd_req;
-	struct ceph_snap_context *snapc;
-	u64 snap_id;
-
-	rbd_assert(osd_req != NULL);
-
-	snapc = img_request ? img_request->snapc : NULL;
-	snap_id = img_request ? img_request->snap_id : CEPH_NOSNAP;
-	osd_req->r_mtime = CURRENT_TIME;
-}
-
 static void rbd_osd_req_format_read(struct rbd_obj_request *obj_request)
 {
 	struct ceph_osd_request *osd_req = obj_request->osd_req;
@@ -1926,6 +1912,12 @@ static void rbd_osd_req_format_write(struct rbd_obj_request *obj_request)
 	osd_req->r_data_offset = obj_request->offset;
 }
 
+static void rbd_osd_req_format_rw(struct rbd_obj_request *obj_request)
+{
+	rbd_osd_req_format_read(obj_request);
+	rbd_osd_req_format_write(obj_request);
+}
+
 /*
  * Create an osd request.  A read request has one osd op (read).
  * A write request has either one (watch) or two (hint+write) osd ops.
-- 
2.12.3

